resolwe/flow/executors/__init__.py

```diff
@@ -315,7 +330,7 @@ def run(self, data_id, script, verbosity=1):
                             for entity in entities:
                                 entity.collections.add(collection)
 
-            if process_rc == 0:
+            if process_rc == 0 and not self.failed:
```

Comment:
No, executor doesn't have the read access to the database, so this flag has to be used to remember what status has been set.